docs/src/explanations/interfaces-and-connections.md

```diff
@@ -83,8 +83,63 @@ where <> bulk connects interfaces of opposite gender between sibling modules or
 
 Bulk connections connect leaf ports of the same name to each other. If the names do not match or are missing, Chisel does not generate a connection.
 
-Caution: bulk connections should only be used with **directioned elements** (like IOs), and is not magical (e.g. connecting two wires isn't supported since Chisel can't necessarily figure out the directions automatically [chisel3#603](https://github.com/freechipsproject/chisel3/issues/603)).
+```scala mdoc:silent:reset
+import chisel3._
+
+class SimpleLink extends Bundle {
+  val data = Output(UInt(16.W))
+  val valid = Output(Bool())
+  val ready = Input(Bool())
+}
+class PLink extends SimpleLink {
+  val parity = Output(UInt(5.W))
+}
+class FilterIO extends Bundle {
+  val x = Flipped(new PLink)
+  val y = new PLink
+}
+class NotReallyAFilterIO extends Bundle {
+  val x = Flipped(new PLink)
+  val y = new PLink
+  //val z = Output(new Bool())
+}
+class Block extends Module {
+  val io1 = IO(new FilterIO)
+  val io2 = IO(Flipped(new NotReallyAFilterIO))
 
+  io1 <> io2
+}
+```
+Caution: bulk connections should only be used with **directioned elements** (like IOs), and is not magical (e.g. connecting two wires isn't supported since Chisel can't necessarily figure out the directions automatically [chisel3#603](https://github.com/freechipsproject/chisel3/issues/603)).
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.util.DecoupledIO
+class Wrapper extends Module{
```

Comment:
Maybe in this document we should rewrite this complicated example to use the same classes that are already in this document

Reply:
Something like:

```
class Block extends Module {
  val io = IO(new FilterIO)
  val f1 = Module(new Filter)
  val f2 = Module(new Filter)
  f1.io.x <> io.x
 val tmp = Wire(...)
  f1.io.y <> tmp
  tmp <> f2.io.x
  f2.io.y <> io.y
}
```